### PR TITLE
Rust version migration to 1.52.1 and corresponding fixes

### PR DESCRIPTION
**Pull request by eadventurous**
Signed-off-by: Egor Ivkov <e.o.ivkov@gmail.com>

### Description of the Change
Rust version migration to 1.52.1 and corresponding fixes

### Benefits
Staying up to date with the latest compiler.

### Possible Drawbacks 
None found.

### CODE REVIEW

File: iroha_data_model/src/lib.rs [hunk 1]
```diff
@@ -2146,72 +2146,72 @@ pub mod pagination {
         #[test]
         fn empty() {
             assert_eq!(
-                vec![1, 2, 3]
+                vec![1_i32, 2_i32, 3_i32]
```
Comment (i1i1) [inline iroha_data_model/src/lib.rs:2149]: Could we please just allow this lint? It is way annoying to type constants like this. Especially in tests which should be quick and dirty?

Comment (eadventurous) [reply iroha_data_model/src/lib.rs:2149]: I think it is okay to allow it in tests, though I only had to correct a few places. Most are ok as is right now.

Comment (eadventurous) [reply iroha_data_model/src/lib.rs:2149]: I think you can allow it in tests, when you need it.

Comment (i1i1) [reply iroha_data_model/src/lib.rs:2149]: IMO we should generally disable it. It doesn't help reading code at all as you rarely want to know types of constants.

**Review by i1i1 — APPROVED** [review]

**Review by takemiyamakoto — APPROVED** [review]

**Pull request merged**